MVStoreException at accountForRemovedPage

I'm working up a POC using this library for temp storage of large datasets for batch processing. Once the map size gets close to the end it throws this error  when `store.commit();` is called.

`org.h2.mvstore.MVStoreException: java.lang.AssertionError: 1 chunk:4b,block:17a,len:3,liveMax:0,livePages:0,map:2,max:2860,next:17d,pages:4,root:12c00003c985,time:4716,unused:471d,unusedAtVersion:4b,version:4b,toc:2170,occupancy:0f {0, 1, 2, 3} [2.1.214/3] [2.1.214/3]`

Thrown from the highlighted `assert` below.

![image](https://user-images.githubusercontent.com/85881324/204556792-87e6795d-e35f-477d-a676-6e86b521d937.png)


The calling code:

```public  T getNext() {
        try (MVStore store = MVStore.open(DB_FILE_NAME)) {
            if (store.hasMap(MAP_NAME)) {
                MVMap<Integer, T> map = store.openMap(MAP_NAME);
                log.info("Map size: " + map.size());
                if (!map.isEmpty()) {
                    T obj =  map.remove(map.firstKey());
                    store.commit();
                    return obj;
                }
            }
            return null;
        }
    }
```

I'm still working through this, but I'm unsure how to troubleshoot this and was looking for some guidance.    

Thank you for your time and help.

--- a/h2/src/main/org/h2/mvstore/MVStore.java
+++ b/h2/src/main/org/h2/mvstore/MVStore.java
@@ -866,28 +866,18 @@ public class MVStore implements AutoCloseable {
                     !map.isVolatile() &&
                     map.hasChangesSince(lastStoredVersion)) {
                 assert rootReference.version <= version : rootReference.version + " > " + version;
-                Page<?,?> rootPage = rootReference.root;
-                if (!rootPage.isSaved() ||
-                        // after deletion previously saved leaf
-                        // may pop up as a root, but we still need
-                        // to save new root pos in meta
-                        rootPage.isLeaf()) {
-                    changed.add(rootPage);
-                }
+                // simply checking rootPage.isSaved() won't work here because
+                // after deletion previously saved page
+                // may pop up as a root, but we still need
+                // to save new root pos in meta
+                changed.add(rootReference.root);
             }
         }
         RootReference<?,?> rootReference = meta.setWriteVersion(version);
         if (meta.hasChangesSince(lastStoredVersion) || metaChanged) {
             assert rootReference != null && rootReference.version <= version
                     : rootReference == null ? "null" : rootReference.version + " > " + version;
-            Page<?, ?> rootPage = rootReference.root;
-            if (!rootPage.isSaved() ||
-                    // after deletion previously saved leaf
-                    // may pop up as a root, but we still need
-                    // to save new root pos in meta
-                    rootPage.isLeaf()) {
-                changed.add(rootPage);
-            }
+            changed.add(rootReference.root);
         }
         return changed;
     }

